examples/samd/interrupt/main.cpp

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2020, Erik Henriksson
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include <modm/board.hpp>
+#include <modm/platform/extint/extint.hpp>
+
+using namespace Board;
+using namespace std::chrono_literals;
+
+static volatile bool blink = true;
+
+void
+isr() {
+  blink = !blink;
+  // Kids, please don't do serial logging in ISRs...
+  MODM_LOG_DEBUG << "blink: " << (blink ? "true" : "false") << modm::endl;
+}
+
+int
+main()
+{
+    Board::initialize();
+    ExternalInterrupt::initialize();
+    ExtInt<3>::initialize(&isr);
+    ExtInt<3>::connectPin<D12>();
```

Comment:
Why not make the interrupt a signal?
```suggestion
    ExternalInterrupt::connect<D12::ExtInt3>(&isr);
```

Reply:
I guess you could even make it a variadic template so that you can attach multiple handlers to the same pin:
```suggestion
    ExternalInterrupt::connect<D12::ExtInt3>(&isr1, &isr2, &isr3);
```
or multiple pins to the same handler?
```suggestion
    ExternalInterrupt::connect<D10::ExtInt1, D11::ExtInt2, D12::ExtInt3>(&isr);
```
It's all just syntactic sugar though…

Reply:
1. I did not do that because I would need code generation to understand what the offsets of "ExtInt3" is to the interrupt registers. Using C++ templates and instance in integer form I can achieve the same thing without code generation.

2. This would slow down interrupt handling for all use cases as the list of handlers get larger. There is also a fairly reasonable workaround, which is very efficient as all those handlers can get inlined into one:

```
void isr() {
  isr1();
  isr2();
  // ...
}
ExtInt<3>::initialize<D12>(&isr);
```

3. You can do this already by calling the same ISR handler on multiple ExtInt<> instances, however, care needs to be taken as the interrupt config is on a signal level, so you would also need to setup the trigger condition N times.

What does make some sense though is to attach multiple pins to the same ExtInt instance, which is already supported:

```
ExternalInterrupt::initialize();
using ButtonInt = ExtInt<3>;
ButtonInt::initialize(&isr, Gpio::InputTrigger::FallingEdge);
ButtonInt::connectPin<GpioA03>();
ButtonInt::connectPin<GpioA19>();
ButtonInt::connectPin<GpioB03>();
```

This won't compile if not all GPIO pins are internally wired to the same ExtInt pin. I guess an improvement in your direction would be to make ExtInt::connectPin variadic though, to replace the last 3 lines with one, WDYT?

Reply:
I think these are some good points, so let's leave it like that.

> Using C++ templates and instance in integer form I can achieve the same thing without code generation.

If all you have is lbuild, everthing looks like code generation… 😅